src/Forms/DefaultFormFactory.php

```diff
@@ -96,14 +97,15 @@ protected function getFormActions(RequestHandler $controller = null, $name, $con
      */
     protected function getFormValidator(RequestHandler $controller = null, $name, $context = [])
     {
-        $validator = null;
-        if ($context['Record']->hasMethod('getCMSValidator')) {
-            // @todo Deprecate or formalise support for getCMSValidator()
-            $validator = $context['Record']->getCMSValidator();
+        if (!$context['Record'] instanceof DataObject) {
+            return null;
         }
 
+        $validator = $context['Record']->getValidatorList();
+
         // Extend validator
         $this->invokeWithExtensions('updateFormValidator', $validator, $controller, $name, $context);
```

Comment:
I think this will need to be amended slightly - currently `$validator` is a single validator, not a list. I think the safest way is probably just to do:

```php
foreach ($validator->getValidators() as $validatorListItem) {
    $this->invokeWithExtensions('updateFormValidator', $validatorListItem, $controller, $name, $context);
}
```

Reply:
For BC concerns, you're right, we'll need to continue to send what was previously being sent.

I would be keen to add in a second extension point as well - one for `updateValidatorList`. I'm not entirely sure if I can reasonably throw a deprecation notice on the old one, but at the very least, I can add some comments around it.